src/main/java/hudson/plugins/mercurial/HgExe.java

```diff
@@ -391,13 +404,19 @@ public String config(FilePath repository, String name) throws IOException, Inter
     /**
      * Runs the command and captures the output.
      */
+    @NonNull
     public String popen(FilePath repository, TaskListener listener, boolean useTimeout, ArgumentListBuilder args)
             throws IOException, InterruptedException {
         args = seed(false).add(args.toCommandArray());
 
         ByteArrayOutputStream data = new ByteArrayOutputStream();
         if (joinWithPossibleTimeout(launch(args).pwd(repository).stdout(data), useTimeout, listener) == 0) {
-            return data.toString();
+            try {
+                //TODO: consider using another charset
+                return data.toString(Charset.defaultCharset().name());
+            } catch (UnsupportedCharsetException ex) { // Should never happen
+                throw new IOException("Cannot perform a conversion using the default charset", ex);
+            }
```

Comment:
Seems silly. Better to just use plain `toString()`, which would use the default charset anyway, and suppress the warning with a comment that this is intentional.